File: ios/Classes/DevrevPlugin.swift
import Flutter
import UIKit
import DevRevSDK
import SessionAnalytics
  
public class DevrevPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "devrev", binaryMessenger: registrar.messenger())
    let instance = DevrevPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
        case "initializeDevRev":
            if let args = call.arguments as? [String: Any],
               let appId = args["appId"] as? String {
                // Pass the API key to the DevRevSDK
                DevRev.configure(appID: appId)
                result("DevRev initialized with API key: \(appId)")
            } else {
                result(FlutterError(code: "INVALID_ARGUMENTS", message: "Missing or invalid arguments", details: nil))
            }
        case "identifyUnverifiedUser":
            Task {
                    do {
                        if let args = call.arguments as? [String: Any],
                            let userId = args["userId"] as? String {
                            try await DevRev.identifyUnverifiedUser(Identity(userID: userId))
                            result("User identified with ID: \(userId)")
                        } else {
                            result(FlutterError(code: "INVALID_ARGUMENTS", message: "Missing or invalid arguments", details: nil))
                        }
                    } catch {
                        result(FlutterError(code: "SHOW_SUPPORT_ERROR", message: "Failed to show support", details: error.localizedDescription))
                    }
                }
        case "identifyAnonymousUser":
            Task {
                do {
                    if let userId = call.arguments as? String {
                        try await DevRev.identifyAnonymousUser(Identity(userID: userId))
                        result("User identified with ID: \(userId)")
                    } else {
                        result(FlutterError(code: "INVALID_ARGUMENTS", message: "Missing or invalid arguments", details: nil))
                    }
                } catch {
                    result(FlutterError(code: "SHOW_SUPPORT_ERROR", message: "Failed to show support", details: error.localizedDescription))
                }
            }
        case "showSupport":
            Task {
                do {
                    try await DevRev.showSupport()
                    result("Support screen shown")
                } catch {
                    result(FlutterError(code: "SHOW_SUPPORT_ERROR", message: "Failed to show support", details: error.localizedDescription))
                }
            }
        case "trackEvent":
            Task {
                do {
                    // capture name: string and properties: Map
                    if let args = call.arguments as? [String: Any],
                        let name = args["name"] as? String,
                        let properties = args["properties"] as? [String: Any] {
                        try await DevRev.trackEvent(name: name, properties: properties)
                        result("Event tracked with name: \(name) and properties: \(properties)")
                    } else {
                        result(FlutterError(code: "INVALID_ARGUMENTS", message: "Missing or invalid arguments", details: nil))
                    }
                }
            }
        case "startRecording":
            SessionAnalytics().startRecording()
        case "stopRecording":
            SessionAnalytics().stopRecording()
        case "pauseRecording":
            SessionAnalytics().pauseRecording()
        case "resumeRecording":
            SessionAnalytics().resumeRecording()
        default:
            result(FlutterMethodNotImplemented)
        }
  }
}

File: android/src/main/kotlin/com/roshan/devrev/DevrevPlugin.kt
package com.roshan.devrev

import ai.devrev.sdk.DevRev
import ai.devrev.sdk.model.Identity
import ai.devrev.sdk.showSupport
import ai.devrev.sdk.trackEvent
import androidx.annotation.NonNull
import android.content.Context
import android.se.omapi.Session
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** DevrevPlugin */
class DevrevPlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private lateinit var channel : MethodChannel
//  private var activity: Activity? = null
  private var context: Context? = null

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    context = flutterPluginBinding.applicationContext
    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "devrev")
    channel.setMethodCallHandler(this)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    if (call.method == "initializeDevRev") {
      val appId = call.argument<String>("appId")
      DevRev.configure(
        context = context!!,
        appId = appId!!
      )
    } else if (call.method == "identifyUnverifiedUser") {
      val userId = call.argument<String>("userId")
      DevRev.identifyUnverifiedUser(identity = Identity(userId!!))
    } else if (call.method == "identifyAnonymousUser") {
      val userId = call.argument<String>("userId")
      DevRev.identifyAnonymousUser(
          userId = userId
      )
    } else if (call.method == "showSupport") {
      DevRev.showSupport(context!!)
    } else if (call.method == "trackEvent") {
      val name = call.argument<String>("name")
      val properties = call.argument<HashMap<String, String>>("properties")
      DevRev.trackEvent(name = name, properties = properties)
    } else if (call.method =="startRecording") {
      SessionAnalytics().startRecording(context = context!!);
    } else if (call.method == "stopRecording") {
      SessionAnalytics().stopRecording()
    } else if (call.method == "pauseRecording") {
      SessionAnalytics().pauseRecording()
    } else if (call.method == "resumeRecording") {
      SessionAnalytics().resumeRecording()
    }
    else {
      result.notImplemented()
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    channel.setMethodCallHandler(null)
  }
}
